plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -679,6 +679,138 @@ SELECT
+	SUM(rs.avg_query_max_used_memory * rs.count_executions) / SUM(rs.count_executions) AS avg_query_max_used_memory,
+	MAX(rs.max_duration) as [max_duration],
+	SUM(rs.avg_duration * rs.count_executions) / SUM(rs.count_executions) as avg_duration,
+	MAX(rs.max_num_physical_io_reads) as max_num_physical_io_reads,
+	SUM(rs.avg_num_physical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_num_physical_io_reads,
+	MAX(rs.max_page_server_io_reads) as max_page_server_io_reads,
+	SUM(rs.avg_page_server_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_page_server_io_reads,
+	MAX(rs.max_log_bytes_used) as [max_log_bytes_used],
+	SUM(rs.avg_log_bytes_used * rs.count_executions) / SUM(rs.count_executions) as avg_log_bytes_used
+FROM sys.query_store_query q WITH (NOLOCK)
+	JOIN sys.query_store_plan p WITH (NOLOCK) ON q.query_id = p.query_id
+	JOIN sys.query_store_runtime_stats rs WITH (NOLOCK) ON p.plan_id = rs.plan_id
+	JOIN sys.query_store_runtime_stats_interval rsi WITH (NOLOCK) ON rs.runtime_stats_interval_id = rsi.runtime_stats_interval_id
+WHERE rsi.start_time >= @currIntervalStartDate AND rsi.end_time <= @currIntervalEndDate
+GROUP BY rs.plan_id, rs.execution_type, rs.runtime_stats_interval_id;
+
+/* Return the timestamp when the query was run, to be used in the next call */
+SELECT CAST(@currIntervalEndTimestamp AS nvarchar(35)) AS QueryData;
+`
+
+const sqlAzureDBQueryStoreWaitStatistics = sqlAzureDBPartQueryPeriod + `
+
+/*QDS is not enabled on read-only replicas: https://docs.microsoft.com/en-us/sql/t-sql/functions/databasepropertyex-transact-sql?view=sql-server-ver15*/
+IF DATABASEPROPERTYEX(DB_Name(), 'Updatability') = 'READ_ONLY'  
+	RETURN
+
+/* Query the data from query store. Group  by plan_id, runtime_stats_interval_id, execution_type and wait_category, and specified here:
+https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-query-store-wait-stats-transact-sql?view=sql-server-ver15
+*/
+SELECT
+	'sqlserver_azuredb_querystore_wait_stats' AS [measurement],
+	REPLACE(@@SERVERNAME, '\', ':') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(nvarchar(35), MAX(rsi.start_time), 126) AS interval_start_time,
+	CONVERT(nvarchar(35), MAX(rsi.end_time), 126) AS interval_end_time,
+	CAST(MAX(q.query_id) AS nvarchar(20)) AS query_id,
+	CONVERT(nvarchar(20), MAX(q.query_hash), 1) as query_hash,
+	CAST(ws.plan_id AS nvarchar(20)) AS plan_id,
+	CONVERT(nvarchar(20), MAX(p.query_plan_hash), 1) as query_plan_hash,
+	ws.wait_category_desc AS wait_category_s,
+	ws.execution_type AS exec_type_d,
+	MAX(ws.execution_type_desc) AS exec_type_desc,
+	SUM(ws.total_query_wait_time_ms) AS total_query_wait_time_ms_d,
+	CAST(SUM(IIF(avg_query_wait_time_ms = 0, 0, ws.total_query_wait_time_ms / avg_query_wait_time_ms)) AS BIGINT) AS count_executions,
+	MAX(ws.max_query_wait_time_ms) AS max_query_wait_time_ms_d,
+	MAX(q.query_parameterization_type) AS query_param_type_d
```

Comment:
Curious, why do we need parameterization type with waits?

Reply:
Parameterization was removed